### PR TITLE
⚗️ AB test for class page to collect CTA

### PR DESCRIPTION
**Pull request by AuroraHuang22**
**variant**
網頁擷取_19-4-2023_15398_localhost

**origin**
網頁擷取_19-4-2023_154815_localhost

### CODE REVIEW

File: src/experiments/class-collect-cta.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default {
+  name: 'class-collect-cta',
+  experimentID: 'cljcOGr3RX-jUiSHCaMu-A',
+  isEligible: ({ route }) => {
+    const parts = route.name.split('___');
+    const routeName = parts[0];
```
Comment (nwingt) [inline src/experiments/class-collect-cta.js:6]: I think it is more robust to use `getRouteBaseName()` from context to extract the route name without localization

You may need to inject `getRouteBaseName` from [here](https://github.com/likecoin/liker-land/blob/57c8784c02c880150c75ae10f7c541cb4aa2c6b7/src/mixins/experiment.js#L14) in order to use it

File: src/pages/nft/class/_classId/index.vue [hunk 2]
```diff
@@ -454,7 +423,9 @@ export default {
       logTrackerEvent(
         this,
         'NFT',
-        'NFTCollect(DetailsPageCTA)',
+        `NFTCollect(DetailsPageCTA_${
+          this.isExperimenting ? 'variant' : 'origin'
+        })`,
```
Comment (nwingt) [inline src/pages/nft/class/_classId/index.vue:428]: It is better to use a full path without constructing dynamically

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/mixins/experiment.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ export default function experimentsMixin(
         const { name, $activeVariants } = this.$exp;
         if (
           this.$exp.isEligible &&
-          !this.$exp.isEligible({ route: this.$route })
+          !this.$exp.isEligible({ route: this.getRouteBaseName(this.$route) })
```
Comment (nwingt) [inline src/mixins/experiment.js:14]: should pass `getRouteBaseName` into `isEligible()` as route is an object with query and params

File: src/pages/nft/class/_classId/index.vue [hunk 4]
```diff
@@ -423,14 +423,23 @@ export default {
...
+    handleCollectFromCTA_exp() {
+      logTrackerEvent(
+        this,
+        'NFT',
+        'NFTCollect(DetailsPageCTA_exp)',
+        this.classId,
+        1
+      );
+      return this.handleCollect();
+    },
```
Comment (nwingt) [inline src/pages/nft/class/_classId/index.vue:441]: I think no need to split, this should be handled with Optimize

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/mixins/experiment.js [hunk 5]
```diff
@@ -11,12 +11,18 @@ export default function experimentsMixin(
         const { name, $activeVariants } = this.$exp;
         if (
           this.$exp.isEligible &&
-          !this.$exp.isEligible({ route: this.$route })
+          !this.$exp.isEligible({
+            route: {
+              name: this.getRouteBaseName(this.$route),
+              query: this.$route.query,
+              params: this.$route.query,
+            },
```
Comment (nwingt) [inline src/mixins/experiment.js:19]:
```suggestion
            route: {
              name: this.getRouteBaseName(this.$route),
              query: this.$route.query,
              params: this.$route.params,
            },
```
I think this is more simple
```suggestion
            route: {
              ...this.$route,
              name: this.getRouteBaseName(this.$route),
            },
```

**Review by nwingt — APPROVED** [review]

File: src/components/NFTCover.vue [hunk 6]
```diff
@@ -23,7 +23,7 @@
       @error="handleImageError"
     >
     <img
-      v-if="!isShowVideo && (!isShowImage || !isLoaded)"
+      v-if="(!isShowVideo && !isShowImage) || !isLoaded"
```
Comment (williamchong) [inline src/components/NFTCover.vue:26]: Is this diff intentional?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/NFTCover.vue (hunk 6 above)
Comment (AuroraHuang22) [reply src/components/NFTCover.vue]: Oh, that was unintentional

**Review by williamchong — APPROVED** [review]

**Pull request merged**